src/transport/SecureMessageCodec.cpp

```diff
@@ -43,22 +43,27 @@ CHIP_ERROR Encrypt(const CryptoContext & context, CryptoContext::ConstNonceView
     VerifyOrReturnError(!msgBuf->HasChainedBuffer(), CHIP_ERROR_INVALID_MESSAGE_LENGTH);
     VerifyOrReturnError(msgBuf->TotalLength() <= kMaxAppMessageLen, CHIP_ERROR_MESSAGE_TOO_LONG);
 
-    static_assert(std::is_same<decltype(msgBuf->TotalLength()), uint16_t>::value,
+    static_assert(std::is_same<decltype(msgBuf->TotalLength()), size_t>::value,
                   "Addition to generate payloadLength might overflow");
```

Comment:
This static assert makes no sense.  It _used_ to make sense because if TotalLength() is 16-bit and tagLen is 16-bit then the addition promotes to 32-bit signed int on 32-bit-or-more systems and can't possibly overflow.

But in the new setup, what does TotalLength being a size_t tell us about overflow?  Absolutely nothing.

Reply:
Agreed that the static_assert does not make sense anymore. Will remove it.